packages/lib/src/components/internal/UIElement/UIElement.tsx

```diff
@@ -384,6 +395,31 @@ export abstract class UIElement<P extends UIElementProps = UIElementProps> exten
         this.handleSuccessResult(response);
     }
 
+    protected handleKeyPress(e: h.JSX.TargetedKeyboardEvent<HTMLInputElement>) {
+        if (e.key === 'Enter' || e.code === 'Enter') {
+            e.preventDefault(); // Prevent <form> submission if Component is placed inside a form
+
+            this.debounceEnterKeyPress({ component: null, fieldType: (e.target as HTMLInputElement).name, action: 'enterKeyPressed' });
+        }
+    }
+
+    /**
+     * Handle Enter key pressed either from a UIElement (in which case this function has been called from handleKeyPress)
+     * OR from a securedField (in which case this function has been called 'directly')
+     * @param obj
+     */
+    protected onEnterKeyPressed(obj: OnKeyPressObj, compRef = this) {
+        // Add component ref here, rather than in handleKeyPress in case this function has been called directly from a securedField
+        obj.component = compRef;
+
+        if (this.props.onEnterKeyPressed) {
+            this.props.onEnterKeyPressed(obj);
+        } else {
+            compRef?.payButtonRef?.buttonElRef?.focus();
```

Comment:
Couldn't we call `this.submit();` directly when the `showPayButton` is `true` 🤔?

Reply:
No, for reasons I tried to explain in the Tech talk, last week. (It's probably easier to demo the "why" rather than explain it!)

Reply:
This would also benefit merchants that are programmatically calling the `component.submit()`, no?

Reply:
Fixed! No need for a `payButtonRef` anymore